src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -181,8 +181,6 @@ object Task {
     // Quick access to the underlying instance identifier of the task.
     val instanceId: Instance.Id
 
-    val reservationId: Reservation.Id
```

Comment:
This is the benefit of it. Tasks and thus instances are not bound to a specific reservation id anymore. *However*, reservation id still require an instance id because `OfferMatcherReconciler` use it to construct an instance map for a certain offer.